src/software/field_tests/field_test_fixture.py

```diff
@@ -0,0 +1,411 @@
+
+    def __init__(
+        self,
+        test_name,
+        blue_full_system_proto_unix_io,
+        yellow_full_system_proto_unix_io,
+        gamecontroller,
+        publish_validation_protos=True,
+    ):
+        """Initialize the FieldTestRunner
+        
+        :param test_name: The name of the test to run
+        :param blue_full_system_proto_unix_io: The blue full system proto unix io to use
+        :param yellow_full_system_proto_unix_io: The yellow full system proto unix io to use
+        :param gamecontroller: The gamecontroller context managed instance 
+        :param publish_validation_protos: whether to publish validation protos
+
+        """
+
+        super(FieldTestRunner, self).__init__(
+            test_name,
+            blue_full_system_proto_unix_io,
+            yellow_full_system_proto_unix_io,
+            gamecontroller,
+        )
+        self.publish_validation_protos = publish_validation_protos
+
+        logger.info("determining robots on field")
+        # survey field for available robot ids
+        try:
+            world = self.world_buffer.get(block=True, timeout=WORLD_BUFFER_TIMEOUT)
+            logger.info(world)
+            self.initial_world = world
+            self.friendly_robot_ids_field = [
+                robot.id for robot in world.friendly_team.team_robots
+            ]
+            self.enemy_robot_ids_field = [
+                robot.id for robot in world.enemy_team.team_robots
+            ]
+
+            logger.info(f"blue team ids {self.friendly_robot_ids_field}")
+
+            if len(self.friendly_robot_ids_field) == 0:
+                raise Exception("no friendly robots found on field")
+
+        except queue.Empty as empty:
+            raise Exception("unable to determine robots on the field")
+
+    def set_tactics(
+        self, tactics: AssignedTacticPlayControlParams, isBlue,
+    ):
+        if isBlue:
+            self.blue_full_system_proto_unix_io.send_proto(
+                AssignedTacticPlayControlParams, tactics
+            )
+        else:
+            self.yellow_full_system_proto_unix_io.send_proto(
+                AssignedTacticPlayControlParams, tactics
+            )
+
+    def set_play(self, play: Play, isBlue):
+        if team == proto.ssl_gc_common_pb2.Team.BLUE:
+            self.blue_full_system_proto_unix_io.send_proto(Play, play)
+
+        else:
+            self.yellow_full_system_proto_unix_io.send_proto(Play, play)
+
+    def send_gamecontroller_command(
+        self,
+        gc_command: proto.ssl_gc_state_pb2.Command,
+        team: proto.ssl_gc_common_pb2.Team,
+        final_ball_placement_point=None,
+    ):
+
+        self.gamecontroller.send_ci_input(
+            gc_command=gc_command,
+            team=team,
+            final_ball_placement_point=final_ball_placement_point,
+        )
+
+    def set_worldState(self, worldstate: WorldState):
+        def __validate(eventually_validation_set, timeout):
```

Comment:
nit: can we add brief javadoc (docstring?) at the beginning of the methods in this file (a couple in the other files are also missing)? The parameters are mostly self explanatory, but the methods themselves can be complex and having a brief description could be very helpful.